exercises/custom-set/example.rs

```diff
@@ -1,149 +1,100 @@
-    pub fn remove(&mut self, needle: &T) -> bool {
-        let mut index: Option<usize> = None;
-        for (i, element) in self.iter().enumerate() {
-            if needle == element {
-                index = Some(i);
-                break;
-            }
-        }
-        match index {
-            Some(index) => {
-                self.elements.remove(index);
-                true
-            }
-            None => false
-        }
+    pub fn is_empty(&self) -> bool {
+        self.collection.is_empty()
     }
-    
-    pub fn difference(&self, other: &CustomSet<T>) -> CustomSet<T> {
-        let mut difference: CustomSet<T> = CustomSet::new();
-        for element in self.elements.iter() {
-            if !other.contains(element) {
-                difference.insert(element.clone());
-            }
-        }
-        difference
+
+    pub fn size(&self) -> usize {
+        self.collection.len()
     }
 
-    pub fn intersection(&self, other: &CustomSet<T>) -> CustomSet<T> {
-        let mut intersection: CustomSet<T> = CustomSet::new();
-        for element in self.elements.iter() {
-            if other.contains(element) {
-                intersection.insert(element.clone());
-            }
-        }
-        intersection
+    pub fn is_subset(&self, other: &Self) -> bool {
+        self.collection.iter().all(|x| other.contains(x))
     }
 
-    pub fn union(&self, other: &CustomSet<T>) -> CustomSet<T> {
-        let mut union = CustomSet::new();
-        for element in self.elements.iter().chain(other.elements.iter()) {
-            union.insert(element.clone());
-        }
-        union
+    pub fn is_disjoint(&self, other: &Self) -> bool {
+        !self.collection.iter().any(|x| other.contains(x))
     }
 
-    pub fn is_disjoint(&self, other: &CustomSet<T>) -> bool {
-        for element in self.elements.iter() {
+    pub fn intersection(&self, other: &Self) -> CustomSet<T> {
+        let mut s = Vec::new();
+        for element in self.collection.iter() {
             if other.contains(element) {
-                return false;
+                s.push(element.clone());
             }
         }
-        true
+        CustomSet::new(s)
     }
 
-    pub fn is_subset(&self, other: &CustomSet<T>) -> bool {
-        for element in self.elements.iter() {
-            if !other.contains(element) {
-                return false;
-            }
+    pub fn union(&self, other: &Self) -> CustomSet<T> {
+        let mut s = Vec::new();
+        for element in self.collection.iter() {
+            s.push(element.clone());
         }
-        true
-    }
-    
-    pub fn is_superset(&self, other: &CustomSet<T>) -> bool {
-        other.is_subset(self)
-    }
-
-    pub fn clear(&mut self) {
-        self.elements.clear();
-    }
-
-    pub fn iter(&self) -> Iter<T>{
-        Iter { iter: self.elements.iter() }
+        for element in other.collection.iter() {
+            s.push(element.clone());
+        }
+        CustomSet::new(s)
```

Comment:
similarly, it may be possible to use https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.chain here, though not strictly necessary